src/SFML/Audio/InputSoundFile.cpp

```diff
@@ -195,15 +196,46 @@ unsigned int InputSoundFile::getSampleRate() const
 ////////////////////////////////////////////////////////////
 Time InputSoundFile::getDuration() const
 {
-    return seconds(static_cast<float>(m_sampleCount) / m_channelCount / m_sampleRate);
+    // Make sure we don't divide by 0
+    if (m_channelCount != 0 && m_sampleRate != 0)
+        return seconds(static_cast<float>(m_sampleCount) / m_channelCount / m_sampleRate);
+    else
+        return Time::Zero;
+}
+
+
+////////////////////////////////////////////////////////////
+Time InputSoundFile::getTimeOffset() const
+{
+    // Make sure we don't divide by 0
+    if (m_channelCount != 0 && m_sampleRate != 0)
+        return seconds(static_cast<float>(m_sampleOffset) / m_channelCount / m_sampleRate);
+    else
+        return Time::Zero;
+}
+
+
+////////////////////////////////////////////////////////////
+Uint64 InputSoundFile::getSampleOffset() const
+{
+    return m_sampleOffset;
 }
 
 
 ////////////////////////////////////////////////////////////
 void InputSoundFile::seek(Uint64 sampleOffset)
 {
     if (m_reader)
-        m_reader->seek(sampleOffset);
+    {
+        // The reader handles an overrun gracefully, but we
+        // pre-check to keep our known position consistent.
+        if (sampleOffset > m_sampleCount)
+            m_sampleOffset = m_sampleCount;
+        else
+            m_sampleOffset = sampleOffset;
```

Comment:
When I wrote that I was unsure whether all c++03 implementations supported uint64 as the parameter type.

But looking at it now, the standard says it's been templatized since at least c++98, so I'll go and make the switch.